sacc2/sacc.py

```diff
@@ -0,0 +1,1026 @@
+        If you just have, for example the z and n(z) data then 
+        use the add_tracer method instead.
+
+        Parameters
+        ----------
+
+        tracer: Tracer instance
+            The tracer object to add to the data set
+        """
+        self.tracers[tracer.name] = tracer
+
+    def add_data_point(self, data_type, tracers, value, tracers_later=False, **tags):
+        """
+        Add a data point to the set.
+
+        Parameters
+        ----------
+        data_type: str
+
+        tracers: tuple of str
+            Strings corresponding to zero or more of the tracers in the data set
+            These should either be already set up using the add_tracer method,
+            or you could set tracers_laster=True if you want to add them later.
+            e.g. for 2pt measurements the tracers are the names of the two n(z)
+            samples
+
+        value: float
+            A single value for the data point
+
+        tracers_later: bool
+            If True, do not complain if the tracers are not know already
+
+        **tags:
+            Tags to apply to this data point.
+            Tags can be any arbitrary metadata that you might want later,
+            For 2pt data the tag would include an angle theta or ell.
+
+        Returns
+        -------
+        None
+        """
+        if self.covariance is not None:
+            raise ValueError("You cannot add a data point after setting the covariance")
+        tracers = tuple(tracers)
+        for tracer in tracers:
+            if (tracer not in self.tracers) and (not tracers_later):
+                raise ValueError(f"Tracer named '{tracer}' is not in the known list of tracers."
+                    "Either put it in before adding data points or set tracers_later=True")
+        d = DataPoint(data_type, tracers, value, **tags)
+        self.data.append(d)
+
+    def add_covariance(self, covariance):
+        """
+        Once you have finished adding data points, add a covariance
+        for the entire set.
+
+        Parameters
+        ----------
+        covariance: array or list
+            2x2 numpy array containing the covariance of the added data points
+            OR a list of blocks
+
+        Returns
+        -------
+        None
+        """
+        if isinstance(covariance, BaseCovariance):
+            self.covariance = covariance
+        else:
+            self.covariance = BaseCovariance.make(covariance, len(self))
+
+
+    def cut(self, mask):
+        """
+        Remove data points and corresponding covariance elements following a mask, in-place.
+
+        You can find indices (e.g. matching some tag) to remove using the indices method.
+
+        Parameters
+        ----------
+        mask: array or list
+            Mask must be either a boolean array or a list/array of integer indices to remove.
+            if boolean then True means to cut data point and False means to keep it
```

Comment:
This seems exactly the opposite of what one would expect (i.e. mask=1 usually means keep - because you're not masking it). You would also save a couple of `not` operations below.

Reply:
I think I made it this way round because the method was called "cut", so to me that meant the True=cut.  But yes, when you have a mask that normally means the other way around.  I'll make a new method called "mask" and rename the argument in this one, so that we have both options.

Reply:
I've added both variants now.